packages/flutter/lib/src/animation/animations.dart

```diff
@@ -434,11 +426,7 @@ class CurvedAnimation extends Animation<double> with AnimationWithParentMixin<do
 
   /// Cleans up any listeners added by this CurvedAnimation.
   void dispose() {
-    // TODO(polina-c): stop duplicating code across disposables
-    // https://github.com/flutter/flutter/issues/137435
-    if (kFlutterMemoryAllocationsEnabled) {
-      FlutterMemoryAllocations.instance.dispatchObjectDisposed(object: this);
-    }
+    assert(maybeDispatchObjectDisposed(this));
```

Comment:
For my own understanding: Why is this wrapped in an assert (and therefore will only execute in debug builds), but `maybeDispatchObjectCreated` isn't?

Reply:
oh, sorry, fixed